server/src/main/java/io/crate/execution/ddl/tables/RenameColumnTask.java

```diff
@@ -0,0 +1,355 @@
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.IndexTemplateMetadata;
+import org.elasticsearch.cluster.metadata.MappingMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.mapper.DocumentMapper;
+import org.elasticsearch.index.mapper.MapperService;
+
+import io.crate.analyze.AlterTableRenameColumnAnalyzer;
+import io.crate.analyze.AnalyzedAlterTableRenameColumn;
+import io.crate.analyze.ParamTypeHints;
+import io.crate.analyze.expressions.ExpressionAnalysisContext;
+import io.crate.analyze.expressions.ExpressionAnalyzer;
+import io.crate.common.CheckedFunction;
+import io.crate.expression.symbol.Symbol;
+import io.crate.expression.symbol.format.Style;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.cluster.DDLClusterStateTaskExecutor;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.metadata.doc.DocTableInfoFactory;
+import io.crate.sql.ExpressionFormatter;
+import io.crate.sql.parser.SqlParser;
+import io.crate.sql.tree.Expression;
+import io.crate.sql.tree.ExpressionRewriter;
+import io.crate.sql.tree.QualifiedNameReference;
+import io.crate.sql.tree.SubscriptExpression;
+
+public class RenameColumnTask extends DDLClusterStateTaskExecutor<RenameColumnRequest> {
+
+    private final NodeContext nodeContext;
+    private final CheckedFunction<IndexMetadata, MapperService, IOException> createMapperService;
+
+    public RenameColumnTask(NodeContext nodeContext,
+                            CheckedFunction<IndexMetadata, MapperService, IOException> createMapperService) {
+        this.nodeContext = nodeContext;
+        this.createMapperService = createMapperService;
+    }
+
+    @Override
+    protected ClusterState execute(ClusterState currentState, RenameColumnRequest request) throws Exception {
+        var renameColumn = request.renameColumn();
+        var relationName = request.relationName();
+
+        var docTableInfoFactory = new DocTableInfoFactory(nodeContext);
+        DocTableInfo docTableInfo = docTableInfoFactory.create(relationName, currentState.metadata());
+        var txnCtx = CoordinatorTxnCtx.systemTransactionContext();
+        ExpressionAnalyzer expressionAnalyzer = new ExpressionAnalyzer(
+            txnCtx, nodeContext, ParamTypeHints.EMPTY, new AlterTableRenameColumnAnalyzer.FieldProviderResolvesUnknownColumns(docTableInfo), null);
+        ExpressionAnalysisContext expressionAnalysisContext = new ExpressionAnalysisContext(txnCtx.sessionSettings());
```

Comment:
Could you try to re-implement this on top of https://github.com/crate/crate/pull/15055 by adding a `DocTableInfo.rename(from, to)` and re-using the `DocTableInfo.writeTo()` method?

Reply:
I think you can ignore that for now as it shouldn't cause test failures. As far as I can tell it's triggered by the templateUpgradeService - which triggers due to the version increment/change.

I'm not sure yet what's the right fix, but the templateUpgradeService probably shouldn't trigger validation of private settings

Reply:
Should be fixed with https://github.com/crate/crate/pull/15141

Reply:
Thank you!